File: 프로그래머스/0/120924. 다음에 올 숫자/다음에 올 숫자.java
class Solution {
    public int solution(int[] common) {
        int answer = 0;
        for(int i =0; i<common.length; i++){
            if((common[2] - common[1]) == (common[1] - common[0])  ){
            answer = common[common.length -1]+(common[2] - common[1]);
        }else if((common[2] / common[1]) == (common[1] / common[0])){
            answer = common[common.length -1]*(common[1] / common[0]);
        }
        }
        
        return answer;
    }
}

File: 프로그래머스/0/120924. 다음에 올 숫자/다음에 올 숫자.js
function solution(common) {
    let answer = 0;
    let a = common[common.length-1];
    let b = common[common.length-2];
    let c = common[common.length-3];
    
    if(a-b === b-c){
        answer = a + (a-b);
    }else if(a/b === b/c){
        answer = a*(a/b);
    }
    return answer;
}